graphql_compiler/tests/integration_tests/test_backends_integration.py

```diff
@@ -426,6 +426,92 @@ def test_recurse(self) -> None:
                     {"descendant_name": "Animal 4"},  # depth 3 match
                 ],
             ),
+            # Query 8: Skip depth 0
+            (
+                """
+            {
+                Animal {
+                    name @filter(op_name: "=", value: ["$starting_animal_name"])
+                    out_Animal_ParentOf {
+                        out_Animal_ParentOf @recurse(depth: 2){
+                            name @output(out_name: "descendant_name")
+                        }
+                    }
+                }
+            }""",
```

Comment:
nit: Indent all of this in one step, to match the indentation of the starting `"""`.